optimization/tests/test_car_model.py

```diff
@@ -68,13 +68,12 @@ def test_energy_used(car_min_0_max_1000, car_min_15_max_45):
 
 def test_energy_used_errors(car_min_0_max_1000):
     with pytest.raises(IndexError):
+        # if velocity points array and the elevation points array are not equal, then the test raises a IndexError.
         # velocity profiles too long (8 velocity points, 7 elevation points)
         car_min_0_max_1000.energy_used([5, 7, 9, 13, 9.5, 7, 5, 20], [(-1, 100), (1, 100), (3, 100), (3, 100), (1, 100), (1, 100), (1.5, 100)])
+        # velocity profiles too short (6 velocity points, 7 elevation points)
```

Comment:
Changes look good. Can we separate this into its own `with pytest.raises()` please? If the first method call raises error as expected, we won't reach the second method call.